vault/core.go

```diff
@@ -273,20 +273,26 @@ type CoreConfig struct {
 	CredentialBackends map[string]logical.Factory
 	AuditBackends      map[string]audit.Factory
 	Physical           physical.Backend
-	Logger             *log.Logger
-	DisableCache       bool   // Disables the LRU cache on the physical backend
-	DisableMlock       bool   // Disables mlock syscall
-	CacheSize          int    // Custom cache size of zero for default
-	AdvertiseAddr      string // Set as the leader address for HA
-	DefaultLeaseTTL    time.Duration
-	MaxLeaseTTL        time.Duration
+
+	// Defaults to the same backend as Physical. This is not a backend that
+	// necessarily supports HA; it is merely the one that will be attempted
+	// for HA operations
+	HAPhysical physical.Backend
```

Comment:
This should be a `physical.HABackend` to simplify things for us in the implementation, otherwise we're casting or checking everywhere. We can make it the consumer's configuration problem to make sure it is (the server command).